frontend/src/modules/EditZing/Components/EmailModal.tsx

```diff
@@ -1,25 +1,92 @@
+// external imports
+import { useState } from 'react'
+import { Box, Button, Typography } from '@mui/material'
+import SendIcon from '@mui/icons-material/Send'
+// zing imports
+import { getBody } from '../utils/emailTemplates'
 import { ZingModal } from '@core/Components'
 import { EmailModalProps } from '../Types/ComponentProps'
-import { useState } from 'react'
 import { TemplateName } from 'EditZing/utils/emailTemplates'
-import { Box, Button, Typography } from '@mui/material'
 import { EmailTemplateButtons } from 'EditZing/Components/EmailTemplateButtons'
 import { EmailPreview } from 'EditZing/Components/EmailPreview'
-import SendIcon from '@mui/icons-material/Send'
+import { sendEmail } from 'Emailing/Components/Emailing'
+import { adminSignIn } from '@fire/firebase'
 
 export const EmailModal = ({
   selectedGroups,
   isEmailing,
   setIsEmailing,
   courseNames,
+  setEmailSent,
+  setEmailSentError,
 }: EmailModalProps) => {
   const [selectedTemplate, setSelectedTemplate] = useState<TemplateName>(
     '' as TemplateName
   )
   const [step, setStep] = useState<number>(0)
-  const titles = ['Select email template', 'Send emails']
+  const titles = [
+    'Select email template',
+    'Send emails',
+    'Auth Error',
+    'Fatal Error',
+  ]
   const title = titles[step]
 
+  // ======= Send Email Helper Functions =======
+
+  /**
+   *
+   * @param group object
+   * @returns the student emails for a given group in a string array.
+   */
+  const groupEmails = async (group: any) => {
+    const rcpts: string[] = ['lscstudypartners@cornell.edu']
+    const members = group.memberData
+    await members.forEach((mem: any) => {
+      rcpts.push(mem.email)
+    })
+    return rcpts
+  }
+
+  /**
+   * promise that sends emails to each group.
+   *
+   * @returns true if there are any errors in sending the email OR false if no errors
+   */
+  const sendGroupEmails = async () => {
+    let failure = false
+    for await (const group of selectedGroups) {
+      const emailRcpts = await groupEmails(group)
+      const emailBody = getBody(selectedTemplate, courseNames.join(', '))
+      const emailSubject = 'Study Partners!'
+      const emailItems = { emailSubject, emailRcpts, emailBody }
+      const emailingResult = await sendEmail(emailItems)
+      if (emailingResult === false) {
+        failure = true
+        console.log(`"made it here" : ${failure}`)
+      }
+    }
+    return failure
+  }
```

Comment:
From what I read as I was looking up for-await trying to figure out what this new syntax is, I think it's meant for asynchronous iterators? I think this would be the cleanest to read if `Promise.all` was used instead so the emails can all be sent at the same time. Also then you can change sendEmail promise to reject if the emailing fails, and put a catch on the `Promise.all`, which will reject if any of the constituent promises reject.